### DISCUSSION TITLE
Non free-text combo behavior when typing free-text

### DISCUSSION THREAD

**Discussion by sandrods** [josefarias/hotwire_combobox#70] [Q&A] [answered]
Hi everyone
 
I've noticed an issue with our comboboxes that aren't configured for free-text (i.e., without the 'name_when_new' option). Despite this configuration, users are still able to type whatever they want into the combobox, resulting in the form being submitted with a nil value. While this behavior is technically correct, it feels counterintuitive to me.

Users might become confused when they receive a validation error indicating that the field is blank, or worse, the system may silently save a nil value if the column is not required.

Do you think it would be beneficial for us to clear the typed text in these cases (perhaps on blur or another trigger) to make it more obvious what is actually happening? Or am I missing something here?

**Comment by josefarias** [comment, marked as answer]
Hey @sandrods! That's a good point. Clearing on blur is interesting. I could get behind that.

If anyone has any concerns (e.g. a11y or otherwise) please share them before we move forward with this.

**Reply by josefarias** [reply]
Implementing as part of https://github.com/josefarias/hotwire_combobox/pull/75

**Comment by sandrods** [comment]
Awesome. Just tested. Worked like a charm
